404.html

```diff
@@ -9,62 +9,192 @@
   </script>
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <meta property="og:title" content="Page not found">
+  <link rel="stylesheet" href="/styles/base.css">
+  <link rel="stylesheet" href="/styles/styles.css">
+  <link rel="stylesheet" href="/styles/global-blocks.css">
   <script src="/scripts/scripts.js" type="module" crossorigin="use-credentials"></script>
   <script type="module">
-    window.addEventListener('load', () => {
-      if (document.referrer) {
-        const { origin, pathname } = new URL(document.referrer);
-        if (origin === window.location.origin) {
-          const backBtn = document.createElement('a');
-          backBtn.classList.add('button', 'error-button-back');
-          backBtn.href = pathname;
-          backBtn.textContent = 'Go back';
-          backBtn.title = 'Go back';
-          const btnContainer = document.querySelector('.button-container');
-          btnContainer.append(backBtn);
-        }
-      }
-    });
-  </script>
-  <script type="module">
-    import { sampleRUM } from '/scripts/aem.js';
-    sampleRUM('404', { source: document.referrer });
+    const getRandom404Data = async () => {
+      try {
+        const res = await fetch("/query-index.json?sheet=404");
+
+        if (!res.ok) {
+          console.warn("We could not load the random 404 image data.");
+          return;
+        };
+
+        const { data } = await res.json();
+
+        if (!data.length) {
+          console.warn("No 404 files found.");
+          return;
+        };
+
+        const random404Item = data[Math.floor(Math.random() * data.length)];
+        const random404Data = {
+          imageSrc: random404Item.image,
+          caption: random404Item.title,
+        };
+        return random404Data;
+      } catch (err) {
+        console.warn("We could not load the random 404 image data.");
+      };
+    }
+
+    const build404Page = async () => {
+      const heading = document.querySelector(".not-found__heading");
+      const description = document.querySelector(".not-found__description");
+      const randomImageContainer = document.querySelector(".not-found__image");
+      const randomImagePicture = randomImageContainer.querySelector("img");
+      const randomImageCaption = randomImageContainer.querySelector("figcaption");
+      const buttons = document.querySelectorAll(".not-found .button");
+  
+      if (window.location.pathname.includes("/careers/")) {
+        heading.classList.add("not-found__heading--careers");
+        heading.textContent = "The job listing you are looking for cannot be found or is no longer active.";
+        description.remove();
+        randomImageContainer.remove();
+        buttons[0].setAttribute("href", "/careers/");
```

Comment:
Buttons switch spots on the careers 404. The design has a "refresh" secondary button on the non-careers 404, and I'm not sure what that entails. It'll be a question for Jelly as well.